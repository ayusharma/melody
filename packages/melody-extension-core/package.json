{
  "name": "melody-extension-core",
  "version": "1.2.0-21.2",
  "description": "",
  "main": "./lib/index.js",
  "jsnext:main": "./lib/index.esm.js",
  "scripts": {
    "build": "mkdir lib; SUPPORT_CJS=true rollup -c ../../rollup.config.js -i src/index.js"
  },
  "author": "",
  "license": "Apache-2.0",
  "dependencies": {
    "@babel/template": "7.4.4",
    "@babel/types": "7.4.4",
    "lodash": "^4.12.0",
    "shortid": "^2.2.6"
  },
  "peerDependencies": {
    "melody-idom": "^1.1.0",
    "melody-parser": "^1.1.0",
    "melody-runtime": "^1.1.0",
    "melody-traverse": "^1.1.0",
    "melody-types": "^1.1.0"
  }
}
