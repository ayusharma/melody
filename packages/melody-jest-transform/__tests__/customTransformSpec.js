/**
 * Copyright 2017 trivago N.V.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS-IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { extension as CoreExtension } from 'melody-extension-core';
import idomPlugin from 'melody-plugin-idom';
import { transformer, process } from '../src';
import * as fs from 'fs';
import * as path from 'path';

jest.mock('find-babel-config', () => ({
    sync: jest.fn().mockImplementation(path => ({
        config: {
            presets: [
                [
                    '@babel/preset-env',
                ],
            ],
            // plugins: [],
        },
    })),
}));

describe('Custom transformer', () => {
    const getFixture = twigPath => {
        const fixturePrefix = './__fixtures__/';
        const filePath = path.join(__dirname, fixturePrefix, twigPath);

        return fs.readFileSync(filePath).toString();
    };

    const test = (fixtureName, options, processFn) => {
        const fixture = getFixture(fixtureName);
        const result = processFn
            ? process(fixture, fixtureName)
            : transformer(fixture, fixtureName, options);
        expect(result).toMatchSnapshot();
    };

    it('should work with default babelconfig and melody-plugins', () => {
        test('test.twig');
    });

    it('should work with custom babelConfig and melody-plugins', () => {
        const plugins = [
            idomPlugin,
            {
                ...CoreExtension,
                options: {
                    svgFilePath: 'testFilePath',
                    embedSvgAsMelody: true,
                },
            },
        ];

        const babel = {
            presets: [
                [
                    '@babel/preset-env',
                ],
            ],
            plugins: ['transform-inline-environment-variables'],
        };

        test('test.twig', { plugins, babel });
    });

    it('should throw error if it cannot find a config in regular locations', () => {
        const mockedFindBabel = require('find-babel-config');
        mockedFindBabel.sync.mockImplementationOnce(() => false);
        const regularFixture = getFixture('test.twig');
        expect(() =>
            transformer(regularFixture, 'test.twig')
        ).toThrowErrorMatchingSnapshot();
    });

    it('should return compiled melody code when noBabel set', () => {
        test('test.twig', { noBabel: true });
    });

    it("should transpile with jest's process function", () => {
        // eslint-disable-line quotes
        test('test.twig', {}, true);
    });
});
