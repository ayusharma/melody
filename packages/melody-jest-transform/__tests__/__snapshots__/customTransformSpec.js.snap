// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom transformer should return compiled melody code when noBabel set 1`] = `
"import { text, elementOpen, elementClose } from \\"melody-idom\\";
export const _template = {};

_template.render = function (_context) {
  elementOpen(\\"div\\", null, null);
  text(\\"test\\");
  elementClose(\\"div\\");
};

if (process.env.NODE_ENV !== \\"production\\") {
  _template.displayName = \\"Test\\";
}

export default function Test(props) {
  return _template.render(props);
}"
`;

exports[`Custom transformer should throw error if it cannot find a config in regular locations 1`] = `"Couldn't find .babelrc or babel entry on package.json! You can specify custom config with \\"transformer\\". Please consult documentation."`;

exports[`Custom transformer should transpile with jest's process function 1`] = `undefined`;

exports[`Custom transformer should work with custom babelConfig and melody-plugins 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports[\\"default\\"] = Test;
exports._template = void 0;

var _melodyIdom = require(\\"melody-idom\\");

var _template = {};
exports._template = _template;

_template.render = function (_context) {
  (0, _melodyIdom.elementOpen)(\\"div\\", null, null);
  (0, _melodyIdom.text)(\\"test\\");
  (0, _melodyIdom.elementClose)(\\"div\\");
};

if (\\"test\\" !== \\"production\\") {
  _template.displayName = \\"Test\\";
}

function Test(props) {
  return _template.render(props);
}"
`;

exports[`Custom transformer should work with default babelconfig and melody-plugins 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports[\\"default\\"] = Test;
exports._template = void 0;

var _melodyIdom = require(\\"melody-idom\\");

var _template = {};
exports._template = _template;

_template.render = function (_context) {
  (0, _melodyIdom.elementOpen)(\\"div\\", null, null);
  (0, _melodyIdom.text)(\\"test\\");
  (0, _melodyIdom.elementClose)(\\"div\\");
};

if (process.env.NODE_ENV !== \\"production\\") {
  _template.displayName = \\"Test\\";
}

function Test(props) {
  return _template.render(props);
}"
`;
